### PR TITLE
3IP: Threads V2

### PR DESCRIPTION
**Pull request by zachferland**
~Still things to workout, just wanted to start adding API changes.~ 

~TODO~
~- should probably list permissions/capabilities for each role explicitly, like the list in notion~
~- need to work out thread naming (or access controller addressing) for members/mods etc~
~- going to have threads with no mods still? what is the default? no mods or first creator mod?~
~- add options for how join thread will be called (members, moderator, ...)~

### CODE REVIEW

File: 3IPs/3ip-2.md [hunk 1]
```diff
@@ -58,6 +55,41 @@ This function returns an array of posts made in the thread. It's also possible t
...
+
+```js
+await deletePost(<id>)
+```
+
+**Add a moderator to a thread:**
+Attempts to add a moderator to a thread. Returns id if sucessful, otherwise false.
+
+```js
+await addMod(<id>)
```
Comment (oed) [inline 3IPs/3ip-2.md]: Why use the abbreviated version? Why not `addModerator`?

Comment (zachferland) [reply 3IPs/3ip-2.md]: I have no preference, we can use `addModerator`

**Comment by zachferland** [comment]
> > going to have threads with no mods still? what is the default? no mods or first creator mod?
> 
> I would prefer if they are disabled, so that we avoid unnecessary pains in the future.

Didn't mean in our current model, (all old are disabled) it would be a no mods AC in the new model

File: 3IPs/3ip-2.md (hunk 1 above)
Comment (oed) [reply 3IPs/3ip-2.md]: Ok I think `addModerator` is clearer!

**Comment by oed** [comment]
> Didn't mean in our current model, (all old are disabled) it would be a no mods AC in the new model

Oh I see what you're saying. If no moderator is given we should default to the current user.

File: 3IPs/3ip-2.md (hunk 1 above)
Comment (zachferland) [reply 3IPs/3ip-2.md]: 👍

**Comment by zachferland** [comment]
@oed updated to reflect what was released now

File: 3IPs/3ip-2.md [hunk 2]
```diff
@@ -58,33 +66,74 @@ This function returns an array of posts made in the thread. It's also possible t
 const posts = thread.getPosts()
 ```
 
+**Delete a thread post:**
+Attempts to delete a post with given id. Deletion completes if the user has the capabilities in the access controller to do so.
+
+```js
+await deletePost(<3ID>)
```
Comment (oed) [inline 3IPs/3ip-2.md]: Should be `postId` here?

Comment (zachferland) [reply 3IPs/3ip-2.md]: yup, fixed 👍

**Pull request merged**